src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -66,43 +66,69 @@
   public static final String DTD_FILE_NAME = "game.dtd";
   private final String mapName;
 
-  public GameParser(final String mapName) {
-    this.mapName = Preconditions.checkNotNull(mapName);
+  private GameParser(final String mapName) {
+    this.mapName = mapName;
   }
 
   /**
-   * Parses the XML Document provided by the given InputStream into a DOM.
-   * 
-   * @return The root Element of the DOM
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  private Element parseDom(final InputStream stream) throws SAXException {
-    Preconditions.checkNotNull(stream, "InputStream must be non-null!");
-    return getDocument(stream).getDocumentElement();
+  public static GameData parse(final String mapName, final InputStream stream)
+      throws GameParseException, SAXException, EngineVersionException {
+    return parse(mapName, stream, new AtomicReference<>());
   }
 
   /**
-   * Parses a file into a GameData object.
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @param gameName Receives the game name parsed from the game definition.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  public GameData parse(final InputStream stream, final AtomicReference<String> gameName)
+  public static GameData parse(final String mapName, final InputStream stream, final AtomicReference<String> gameName)
+      throws GameParseException, SAXException, EngineVersionException {
+    checkNotNull(mapName);
+    checkNotNull(stream);
+    checkNotNull(gameName);
+
+    return new GameParser(mapName).parse(stream, gameName);
+  }
+
+  private GameData parse(final InputStream stream, final AtomicReference<String> gameName)
       throws GameParseException, SAXException, EngineVersionException {
     final Element root = parseDom(stream);
     parseMapProperties(root, gameName);
-    // everything until here is needed to select a game
     parseMapDetails(root);
     return data;
   }
 
   /**
-   * Parses just the essential parts of the maps.
-   * Used to display all available maps.
+   * Performs a shallow parse of the game definition contained in the specified stream.
+   *
+   * @return A partial {@link GameData} instance that can be used to display metadata about the game (e.g. when
+   *         displaying all available maps); it cannot be used to play the game.
    */
-  public GameData parseMapProperties(final InputStream stream, final AtomicReference<String> gameName)
+  public static GameData parseShallow(final String mapName, final InputStream stream)
```

Comment:
I really hope we can kill this method, the parse full vs shallow methods seem to not be fully appropriate. Seems like the 'parse' shallow should be done in a different way completely, rather than trying to re-use code in this way. This is not actionable, but do want to caution that this method is very likely to become deprecated.